src/puppetlabs/comidi.clj

```diff
@@ -21,17 +21,24 @@
 (def Zipper
   (schema/pred ks/zipper?))
 
-(def http-methods
-  #{:any :get :post :put :delete :head :options})
-
 (def RequestMethod
   (schema/enum :any :get :post :put :delete :head :options))
 
+; Derived from bidi-schema PatternSegment
 (def RegexPatternSegment
-  [(schema/one Pattern "regex") (schema/one schema/Keyword "variable")])
+  (schema/pair schema/Regex "qual" schema/Keyword "id"))
+
+; Derived from bidi-schema by flattening a vector of Patterns
+(def RouteInfoPathSegment
+  (schema/either bidi-schema/Path
+                 bidi-schema/PatternSegment
+                 schema/Bool))
```

Comment:
@rlinehan and I had a chat there and concluded we could avoid ```either``` *and* ```conditional```, simplify our schema and rely on the clojure (cond ...) to give us reasonable validation.